### PR TITLE
004 view role band levels

### PR DESCRIPTION
**Pull request by Patrick-Mc-Grath**

### CODE REVIEW

**Review by malachi-mccarney — APPROVED** [review]
Looks good

File: SQLScripts/US004 - Populate Bands Table.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+INSERT INTO TABLE Bands(band_name) VALUES('Trainee');
+INSERT INTO TABLE Bands(band_name) VALUES('Apprentice');
+INSERT INTO TABLE Bands(band_name) VALUES('Associate');
+INSERT INTO TABLE Bands(band_name) VALUES('Senior Associate');
+INSERT INTO TABLE Bands(band_name) VALUES('Consultant');
+INSERT INTO TABLE Bands(band_name) VALUES('Manager');
+INSERT INTO TABLE Bands(band_name) VALUES('Principal');
+INSERT INTO TABLE Bands(band_name) VALUES('Leadership');
```
Comment (ccurran01) [inline SQLScripts/US004 - Populate Bands Table.sql:8]: Would something like this work instead? 

```suggestion
INSERT INTO TABLE Bands(band_name) 
VALUES 
     ('Trainee'),
     ('Apprentice'),
     ('Senior Associate'),
     ...
```

File: SQLScripts/US004 - Populate Role_Bands Table.sql
Comment (ccurran01) [inline SQLScripts/US004 - Populate Role_Bands Table.sql:1]: Should this be empty?

File: src/main/java/com/kainos/ea/controller/RoleBandController.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    @GET
+    @Path("/role-band-levels")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getRoleBands() {
+        try {
+            List<RoleBandResponse> roleBandList = roleBandService.getRoleBands();
+            return Response.ok().entity(roleBandList).build();
+        } catch (DatabaseConnectionException | FailedToGetRoleBandsException e) {
```
Comment (ccurran01) [inline src/main/java/com/kainos/ea/controller/RoleBandController.java:45]: What about handling this in the service?

File: SQLScripts/US004 - Populate Role_Bands Table.sql
Comment (Patrick-Mc-Grath) [reply SQLScripts/US004 - Populate Role_Bands Table.sql:1]: It wasn't empty locally, but didn't get pushed for some reason.

**Review by malachi-mccarney — APPROVED** [review]

**Review by ccurran01 — APPROVED** [review]
Looks good @Patrick-Mc-Grath.

**Review by ieuanbyers — APPROVED** [review]

File: src/main/java/com/kainos/ea/dao/JobRoleDao.java [hunk 3]
```diff
@@ -11,27 +12,30 @@
...
+            Statement st = c.createStatement();
 
-        ResultSet rs = st.executeQuery(" SELECT Job_Roles.role_id, Job_Roles.role_title, Job_Roles.job_family_id, Capabilities.name " +
-                                        " FROM Job_Families " +
-                                        " INNER JOIN Job_Roles on Job_Roles.job_family_id = Job_Families.job_family_id " +
-                                        " INNER JOIN Capabilities on Capabilities.capability_id  = Job_Families.capability_id " +
-                                        " ORDER BY Job_Roles.role_id ASC");
-        
-        List<JobRole> jobRoles = new ArrayList<>();
+            ResultSet rs = st.executeQuery(" SELECT Job_Roles.role_id, role_title, Bands.band_name, Job_Families.name, Capabilities.name" +
```
Comment (martin-doherty1) [inline src/main/java/com/kainos/ea/dao/JobRoleDao.java:18]: maybe add an alias for Jobfamily.name and capabilities.name so they don't get confused

**Review by ccurran01 — APPROVED** [review]

**Pull request merged**